doc/Data Plane Code Generation/DeveloperDrivenEvolution.md

```diff
@@ -0,0 +1,195 @@
+```
+- Adding a suffix `Values` to the initial method name as the convenience method name, if the initial method is plural (e.g., change `GetMetricFeedbacksAsync` to `GetMetricFeedbackValuesAsync`).
+```C#
+public virtual async Task<Response<MetricFeedback>> GetMetricFeedbackValuesAsync(string feedId, CancellationToken cancellationToken = default);
+```
+**If this name doesn't make sense, pick the best name**
+
+If the suggested rename does not apply to your scenario, you could pick the best name fitting your API.
+## Create the models
+Currently the models should be created manually. We will support generate it automatically.
+
+After you create the models, you also need to add the helper methods mapping between raw response and model. Examples for model `MetricFeedback` (Models/MetricFeedback/MetricFeedback.cs) are:
+```C#
+namespace Azure.AI.MetricsAdvisor
+{
+    public partial class MetricFeedback
+    {
+        // Mapping raw response to model
+        internal static MetricFeedback FromResponse(Response response)
+        {
+            using var document = JsonDocument.Parse(response.Content);
+            // Add your deserialization logic in DeserializeMetricFeedback
+            return DeserializeMetricFeedback(document.RootElement);
+        }
+
+        // Mapping model to raw request
+        internal static RequestContent ToRequestContent(MetricFeedback metricFeedback)
+        {
+            var content = new Utf8JsonRequestContent();
+            content.JsonWriter.WriteObjectValue(metricFeedback);
+            return content;
+        }
+    }
+}
+```
+## Implement the method
+You could call the protocol method inside your convenience method and map the returned raw response to model. See different scenarios and corresponding examples below. 
+
+### Implement GET method
+
+**Generated code before (Generated/MetricsAdvisorClient.cs):**
+``` C#
+namespace Azure.AI.MetricsAdvisor
+{
+    public partial class MetricsAdvisorClient
+    {
+        public virtual async Task<Response> GetMetricFeedbackAsync(Guid feedbackId, RequestContext context = null)
+        {
+        }
+    }
+}
+```
+**Improve the GET method (MetricsAdvisorClient.cs):**
+``` C#
+namespace Azure.AI.MetricsAdvisor
+{
+    public partial class MetricsAdvisorClient
+    {
+        // Suggest appending value to method name
+        public virtual async Task<Response<MetricFeedback>> GetMetricFeedbackValueAsync(Guid feedbackId, CancellationToken cancellationToken = default)
+        {
+            // Call protocol method
+            Response response = await GetMetricFeedbackAsync(feedbackId, new RequestContext() { CancellationToken = cancellationToken });
+
+            // Calling deserialization helper
+            MetricFeedback value = MetricFeedback.FromResponse(response);
+        }
+    }
+}
+```
+
+### Implement POST method
```

Comment:
This doesn't illustrate that the client API before is:

```csharp
namespace Azure.AI.MetricsAdvisor {
       public class MetricsAdvisorClient {
           // ...
           public virtual async Task<Response> CreateMetricFeedbackAsync(RequestContent content, RequestContext context = null);
           // ...
           }
    }
```

And after the addition of the grow-up method it is:

```csharp
namespace Azure.AI.MetricsAdvisor {
       public class MetricsAdvisorClient {
           // ...
           public virtual async Task<Response> CreateMetricFeedbackAsync(RequestContent content, RequestContext context = null);
public virtual async Task<Response<MetricFeedback>> CreateMetricFeedbackAsync(MetricFeedback feedback, CancellationToken cancellationToken = default);
           // ...
           }
    }
```

Please make this clear.  

The section https://github.com/Azure/autorest.csharp#change-model-namespace-or-accessibility-in-bulk is a nice example of this, with diff-color highlighting

![image](https://user-images.githubusercontent.com/21269347/160900518-a99117c4-599c-405a-b9ec-4cb1dac58325.png)

Reply:
I disagree that this is clear, but we can test it with partner teams referencing the docs and improve if necessary.